Code/Framework/AzCore/Tests/DOM/DomJsonTests.cpp

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/DOM/Backends/JSON/JsonBackend.h>
+#include <AzCore/DOM/Backends/JSON/JsonSerializationUtils.h>
+#include <AzCore/Name/NameDictionary.h>
+#include <AzCore/Serialization/Json/JsonUtils.h>
+#include <AzCore/UnitTest/TestTypes.h>
+
+namespace AZ::DOM::Tests
+{
+    class DomJsonTests : public UnitTest::AllocatorsFixture
+    {
+    public:
+        void SetUp() override
+        {
+            UnitTest::AllocatorsFixture::SetUp();
+            NameDictionary::Create();
+            m_document = AZStd::make_unique<rapidjson::Document>();
+        }
+
+        void TearDown() override
+        {
+            m_document.reset();
+            NameDictionary::Destroy();
+            UnitTest::AllocatorsFixture::TearDown();
+        }
+
+        static bool DeepCompare(const rapidjson::Value& lhs, const rapidjson::Value& rhs)
```

Comment:
The Json Serialization already provides this functionality, see `Compare` in `\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerialization.h`.